Formula/f/flint.rb

```diff
@@ -9,6 +9,7 @@ class Flint < Formula
   livecheck do
     url :stable
     strategy :github_latest
+    regex(/^v?(\d+(?:.\d+)+(-.+)?)$/i)
```

Comment:
```suggestion
    regex(/^v?(\d+(?:\.\d+)+(?:-p\d+)?)$/i)
    strategy :github_latest
```

A few tweaks:

* `#regex` should come before `#strategy` in a `livecheck` block, since we sometimes use a `strategy` block and pass the regex into it.
* The dot between the version parts should be escaped (`\.` instead of `.`), since we only want to match a dot here (`.` in a regex matches anything).
* The flint repository also contains unstable suffixes like `alpha1`, `beta1`, `-rc1`, and `-dev`, so there's something to be said about explicitly matching an optional `-p#` suffix (not any suffix). Unstable releases shouldn't be marked as "latest" (and should be marked as "pre-release") but there's a previous v3.0.0-rc1 relesae that isn't marked as "pre-release", so it would be better for this to temporarily return an error instead of return an unstable version as newest.
* We should use a non-capturing group for the suffix (`(?:)`), since we're only creating a group to make it optional and don't need to capture the matching contents (we only want to capture the version text in this case).

Reply:
> * `#regex` should come before `#strategy` in a `livecheck` block, since we sometimes use a `strategy` block and pass the regex into it.

This should probably be enforced by an audit.